feature(sm): extract oidc from ref arch (#7906)
camunda/camunda-docs: docs/self-managed/deployment/helm/cloud-providers/_partials/_terraform-s3-bucket.md

#### Create an S3 bucket for Terraform state management

Before setting up Terraform, you need to create an S3 bucket that will store the state file. This is important for collaboration and to prevent issues like state file corruption.

To start, set the region as an environment variable upfront to avoid repeating it in each command:

```bash
export AWS_REGION=<your-region>
```

Replace `<your-region>` with your chosen AWS region (for example, `eu-central-1`).

:::note Special case us-east-1

Regions outside of `us-east-1` require the appropriate `LocationConstraint` to be specified in order to create the bucket in the desired region.
While `us-east-1` does not require it and can only be created without specifying it.

:::

> :::note Region of the bucket's state

Author: I agree, it does not looks good, maybe having a `details` for us-east-1 would suffices

> This region can be different from the regions used for other resources, but it requires to be set explicitly in the backend configuration using the flag: `-backend-config="region=<your-region>"`.

Reviewer: ```suggestion
This region can be different from regions used for other resources, but this must be configured explicitly in the backend: `-backend-config="region=<your-region>"`.
```

For clarity, this guide explicitly sets the bucket region in all relevant commands.
:::

Now, follow these steps to create the S3 bucket with versioning enabled:

1. Open your terminal and ensure the AWS CLI is installed and configured.

> 2. Run the following command to create an S3 bucket for storing your Terraform state. Make sure to use a unique bucket name and set the `AWS_REGION` environment variable beforehand:

Author: same rationale as previous comment, coma adds ambiguity

> 2. Run the following command to create an S3 bucket for storing your Terraform state. Make sure to use a unique bucket name and set the `AWS_REGION` environment variable beforehand:

Reviewer: In what way? These are two, independent imperatives:

- Make sure to use a unique bucket name.
- Set the `AWS_REGION` environment variable beforehand.

Because they are independent clauses, they should be separated by a comma. This eliminates the ambiguity of potential overlap. If anything, the `AWS_REGION` imperative should come first, because we explicitly state to do that "beforehand".
